### PR TITLE
Move add/remove logger backend to client

### PR DESCRIPTION
**Pull request by josevalim**
/cc @lexmag @fishcakez

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/server.ex [hunk 1]
```diff
@@ -134,46 +114,27 @@ defmodule ExUnit.Server do
...
-            {:error, reason}
-        after
-          0 -> {:error, :not_found}
-        end
-      :ok -> :ok
+  defp remove_log_capture(ref, %{log_captures: refs} = config) do
+    if ref in refs do
+      refs = HashSet.delete(refs, ref)
+      maybe_add_console(refs, config.log_status)
+      %{config | log_status: nil, log_captures: refs}
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/server.ex]: I think we shouldn't change `log_status` to `nil` as it won't match in `maybe_add_console/2` next time.

Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/server.ex]: Ah, good catch.

File: lib/ex_unit/lib/ex_unit/capture_log.ex [hunk 2]
```diff
@@ -82,4 +82,19 @@ defmodule ExUnit.CaptureLog do
...
     end
   end
+
+  defp add_capture(pid, opts) do
+    GenEvent.add_mon_handler(Logger, {Console, pid}, {pid, opts})
+  end
+
+  defp remove_capture(pid) do
+    case GenEvent.remove_handler(Logger, {Console, pid}, nil) do
+      :ok -> :ok
```
Comment (fishcakez) [inline lib/ex_unit/lib/ex_unit/capture_log.ex]: flush message here too!

File: lib/ex_unit/lib/ex_unit/server.ex [hunk 3]
```diff
@@ -83,40 +81,22 @@ defmodule ExUnit.Server do
...
-    catch
-      :exit, :noproc ->
-        Process.demonitor(ref, [:flush])
-        {:reply, {:error, :no_logger}, config}
+  def handle_call({:log_capture_on, pid}, _from, config) do
+    ref  = Process.monitor(pid)
+    refs = HashSet.put(config.log_captures, ref)
+
+    if HashSet.size(refs) == 1 do
+      status = Logger.remove_backend(:console)
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/server.ex]: If the Logger app is not running the will be an exit.

Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/server.ex]: We won't reach this if the Logger app is not running because add_capture would fail. :)

Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/server.ex]: Ah, right.

Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/server.ex]: Then we probably don't need to return `{:ok, ref}` and use just `ref`.

File: lib/ex_unit/test/ex_unit/capture_log_test.exs [hunk 4]
```diff
@@ -30,6 +30,7 @@ defmodule ExUnit.CaptureLogTest do
 
     # Ensure no leakage on failures
     assert group_leader == Process.group_leader()
+    refute_received {:gen_event_EXIT, _, _}
```
Comment (lexmag) [inline lib/ex_unit/test/ex_unit/capture_log_test.exs]: Will prepare a PR with bunch of tricky tests. :laughing:

**Pull request merged**